cime_config/hist_config.py

```diff
@@ -0,0 +1,1126 @@
+        return add_ok
+
+    def remove_fields(self, fields, pobj, logger):
+        """Remove each field in <fields> from all lists it is on.
+        Return True if each field was found (and removed)."""
+        fields_to_delete = set(fields)
+        fields_deleted = set()
+        all_removed = True
+        for fld_list in self.__all_fields:
+            removed = fld_list.remove_fields(fields_to_delete, pobj, logger)
+            fields_deleted.update(removed)
+        # end for
+        if fields_to_delete != fields_deleted:
+            all_removed = False
+            missing_fields = fields_to_delete.difference(fields_deleted)
+            ctx = context_string(pobj)
+            logger.warning(self.__missing_msg.format(", ".join(list(missing_fields)),
+                                       self.volume, ctx))
+        # end if
+        return all_removed
+
+    @property
+    def volume(self):
+        """Return the volume for this HistoryVolConfig object"""
+        return self.__volume
+
+    @property
+    def precision(self):
+        """Return the precision property for this HistoryVolConfig object"""
+        return self.__precision
+
+    def set_precision(self, prec, pobj, logger):
+        """Modify the precision property of this HistoryVolConfig object.
+        Return True if <prec> is a recognized precision"""
+        if prec.upper() in _OUT_PRECS:
+            self.__precision = prec
+            self.__precision_set = True
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting precision to '{prec}'{ctx}")
+            # end if
+            return True
+        # end if
+        emsg = f"Attempt to set unrecognized precision, '{prec}'"
+        pobj.add_syntax_err(emsg)
+        return False
+
+    @property
+    def max_frames(self):
+        """Return the max_frames property for this HistoryVolConfig object"""
+        return self.__max_frames
+
+    @property
+    def write_nstep0(self):
+        """Return the write_nstep0 property for this HistoryVolConfig object"""
+        return self.__write_nstep0
+
+    def set_max_frames(self, nframes, pobj, logger):
+        """Modify the max_frames property of this HistoryVolConfig object.
+        Return True if <nframes> is a valid setting."""
+        nframes_ok = True
+        nframes_i, _ = _is_integer(nframes)
+        nframes_ok = nframes_i and (nframes > 0)
+        if nframes_ok:
+            self.__max_frames = nframes_i
+            self.__max_frames_set = True
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting max frames to '{nframes}'{ctx}")
+            # end if
+        else:
+            emsg = f"Attempt to set max frames to '{nframes}', must be a positive integer"
+            pobj.add_syntax_err(emsg)
+        # end if
+        return nframes_ok
```

Comment:
```suggestion
            return True
        # end if

        emsg = f"Attempt to set max frames to '{nframes}', must be a positive integer"
        pobj.add_syntax_err(emsg)
        return False
```
Looks a little more readable with explicit return values so it's easier to debug.  Also saves us an else block.

Reply:
updated.